fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/patient/ChunkReader.java

```diff
@@ -0,0 +1,309 @@
+
+                // Todo: uncomment this after the sort lastModified bug is fixed
+                //queryParameters.put("_sort", Arrays.asList(new String[] { Constants.FHIR_SEARCH_LASTUPDATED }));
+
+                searchContext = SearchUtil.parseQueryParameters("Patient", patient.getId(),
+                        ModelSupport.getResourceType(resourceTypes.get(indexOfCurrentResourceType)), queryParameters, null, true);
+                do {
+                    searchContext.setPageSize(pageSize);
+                    searchContext.setPageNumber(compartmentPageNum);
+                    FHIRTransactionHelper txn = new FHIRTransactionHelper(fhirPersistence.getTransaction());
+                    txn.begin();
+                    FHIRPersistenceContext persistenceContext = FHIRPersistenceContextFactory.createPersistenceContext(null, searchContext);
+                    List<Resource> resources2 = fhirPersistence.search(persistenceContext, resourceType).getResource();
+                    txn.commit();
+                    compartmentPageNum++;
+
+                    for (Resource res2 : resources2) {
+                        if (res2 == null) {
+                            continue;
+                        }
+                        try {
+                            FHIRGenerator.generator(Format.JSON).generate(res2, chunkData.getBufferStream());
+                            chunkData.getBufferStream().write(Constants.NDJSON_LINESEPERATOR.getBytes());
+                            resSubTotal++;
+                        } catch (FHIRGeneratorException e) {
+                            if (res.getId() != null) {
+                                logger.log(Level.WARNING, "fillChunkDataBuffer: Error while writing resources with id '"
+                                        + res.getId() + "'", e);
+                            } else {
+                                logger.log(Level.WARNING,
+                                        "fillChunkDataBuffer: Error while writing resources with unknown id", e);
+                            }
+                        } catch (IOException e) {
+                            logger.warning("fillChunkDataBuffer: chunkDataBuffer written error!");
+                            throw e;
+                        }
+                    }
+
+                } while (searchContext.getLastPageNumber() >= compartmentPageNum);
+            }
+            chunkData.setCurrentPartResourceNum(chunkData.getCurrentPartResourceNum() + resSubTotal);
+            logger.info("fillChunkDataBuffer: Processed resources - " + resSubTotal + "; Bufferred data size - "
+                    + chunkData.getBufferStream().size());
+        } else {
+            logger.warning("fillChunkDataBuffer: chunkData is null, this should never happen!");
+            throw new Exception("fillChunkDataBuffer: chunkData is null, this should never happen!");
+        }
+
+    }
+
+    /**
+     * @see AbstractItemReader#readItem()
```

Comment:
We've been removing these types of comments

Reply:
removed